debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,527 @@
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_INTERFACE_INSTANCE_METHOD, "Methodref");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_HANDLE_METHOD, "HandleMethodref");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_INTERFACE_METHOD, "InterfaceMethodref");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_METHOD_TYPE, "MethodType");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_METHODHANDLE, "MethodHandle");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_ANNOTATION_UTF8, "UTF8");
+		cpTypeToString.put(J9ConstantPool.J9CPTYPE_CONSTANT_DYNAMIC, "ConstantDynamic");
+	};
+
+	public J9ConstantPoolCommand() {
+		addCommand("constantpool", "<ramclass>", "dump constant pool for the given RAM class.");
+	}
+
+	public static boolean isSupported() {
+        return (0 != J9VTABLE_INITIAL_VIRTUAL_OFFSET)
+            && (0 != J9Consts.J9_ITABLE_INDEX_METHOD_INDEX)
+            && (0 != J9Consts.J9_ITABLE_INDEX_OBJECT)
+            && (0 != J9Consts.J9_ITABLE_INDEX_SHIFT)
+            && (0 != J9Consts.J9_ITABLE_INDEX_TAG_BITS)
+            && (0 != J9Consts.J9_ITABLE_INDEX_UNRESOLVED)
+            && (0 != J9NonbuilderConstants.J9DescriptionCpBsmIndexMask)
+            && (0 != J9NonbuilderConstants.J9DescriptionCpTypeShift);
+    }
+
+	private static J9ClassPointer findClassByName(J9JavaVMPointer vm, String name) throws CorruptDataException {
+		ClassSegmentIterator iterator = new ClassSegmentIterator(vm.classMemorySegments());
+		while (iterator.hasNext()) {
+			J9ClassPointer classRef = (J9ClassPointer) iterator.next();
+			String className;
+			if (J9ClassHelper.isArrayClass(classRef)) {
+				className = J9ClassHelper.getArrayName(classRef);
+			} else {
+				className = J9ClassHelper.getName(classRef);
+			}
+			if (className.equals(name)) {
+				return classRef;
+			}
+		}
+		return null;
+	}
+
+	private static boolean sameNameAndSignature(J9ROMNameAndSignaturePointer a, J9ROMNameAndSignaturePointer b) throws CorruptDataException {
+		return J9UTF8Helper.stringValue(a.name()).equals(J9UTF8Helper.stringValue(b.name())) &&
+			J9UTF8Helper.stringValue(a.signature()).equals(J9UTF8Helper.stringValue(b.signature()));
+	}
+
+	private static J9RAMClassRefPointer J9VMConstantPoolClassRefAt(J9JavaVMPointer vm, long index) throws CorruptDataException {
+		J9RAMConstantPoolItemPointer jclConstantPool = vm.jclConstantPoolEA();
+		jclConstantPool.addOffset(index);
+		J9RAMClassRefPointer ref = J9RAMClassRefPointer.cast(jclConstantPool.add(index));
+		return ref;
+	}
+
+	private static PointerPointer getSpecialTable(J9ClassPointer ramClass) throws CorruptDataException {
+		try {
+			Lookup lookup = MethodHandles.lookup();
+			Class<?> classPointer = J9ClassPointer.class;
+			MethodHandle mhTable;
+			String methodName = J9BuildFlags.J9VM_OPT_OPENJDK_METHODHANDLE ? "invokeCache" : "methodTypes";
+			mhTable = lookup.findVirtual(classPointer, methodName, MethodType.methodType(PointerPointer.class));
```

Comment:
Please just use `J9ClassPointer.class` instead of `classPointer`.
Please move the declaration of `mhTable` into the assignment.
```
	String methodName = J9BuildFlags.J9VM_OPT_OPENJDK_METHODHANDLE ? "invokeCache" : "methodTypes";
	MethodHandle mhTable = lookup.findVirtual(J9ClassPointer.class, methodName, MethodType.methodType(PointerPointer.class));
```